server/item/recipe/recipe.go

```diff
@@ -31,22 +31,37 @@ func NewShapeless(input []Item, output item.Stack, block string) Shapeless {
 	}}
 }
 
-// Smithing represents a recipe only craftable on a smithing table.
-type Smithing struct {
+// SmithingTransform represents a recipe only craftable on a smithing table.
+type SmithingTransform struct {
 	recipe
 }
 
-// NewSmithing creates a new smithing recipe and returns it. The recipe can only be crafted on the block passed in the
-// parameters. If the block given a crafting table, the recipe can also be crafted in the 2x2 crafting grid in the
-// player's inventory.
-func NewSmithing(base, addition, template Item, output item.Stack, block string) Smithing {
-	return Smithing{recipe: recipe{
+// NewSmithingTransform creates a new smithing recipe and returns it. The recipe can only be crafted on the block
+// passed in the parameters. If the block given a crafting table, the recipe can also be crafted in the 2x2 crafting
+// grid in the player's inventory.
+func NewSmithingTransform(base, addition, template Item, output item.Stack, block string) SmithingTransform {
```

Comment:
The block parameter/the associated comment looks rather out of place, maybe this should be removed?